components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultAuthenticationRequestHandler.java

```diff
@@ -143,7 +144,8 @@ public void handle(HttpServletRequest request, HttpServletResponse response,
 
     private void persistCurrentContext(AuthenticationContext context) {
 
-        FrameworkUtils.addAuthenticationContextToCache(context.getContextIdentifier(), context);
+        long timeoutPeriod = TimeUnit.MINUTES.toNanos(IdentityUtil.getTempDataCleanUpTimeout());
+        FrameworkUtils.addAuthenticationContextToCache(context.getContextIdentifier(), context, timeoutPeriod);
```

Comment:
When expiry time is a property of the cache entry, why that needs to be passed. Shouldn't this be managed within the 'addAuthenticationContextToCache' method which adds the entry to the cache.